src/mc-fle2-insert-update-payload-v2.c

```diff
@@ -444,18 +444,22 @@ SERIALIZE_TEXT_TOKEN_SET_FOR_TYPE_IMPL(Prefix)
         const char *index_string = NULL;                                                                               \
         char storage[16];                                                                                              \
         uint32_t index = 0;                                                                                            \
-                                                                                                                       \
-        for (size_t i = 0; i < array->len; i++) {                                                                      \
-            mc_Text##Type##TokenSet_t ts = _mc_array_index(array, mc_Text##Type##TokenSet_t, i);                       \
+        uint32_t limit = (array->len > UINT32_MAX) ? UINT32_MAX : (uint32_t)array->len;                                \
+        while (index < limit) {                                                                                        \
```

Comment:
Nit: 

```suggestion
        for (; index < limit; index++) {                                                                                        \
        
```
And remove the increment at the end of the loop.

Reply:
done